### DISCUSSION TITLE
GA4GH Refget API

### DISCUSSION THREAD

**Discussion by emiliorighi** [GMOD/jbrowse-components#3412] [Ideas] [answered]
Hi,

I would like to create a custom sequence adapter that consumes the[ refget api](https://www.ebi.ac.uk/ena/cram/swagger-ui/index.html#/The%20GA4GH%20Refget%20reference%20sequence%20retrieval%20API) to retrieve the assembly's sequence.

My idea is to use a text file containing the sequences INSDC accessions and then trigger the call to API based on the user interactions.
In a certain sense is like the bgzip adapter but requesting a range of nucleotides to the API instead of the byte range. 

Is it possible to implement something like this with just a custom sequence adapter? 

I am using the LGV react component in my VueJs application, how can I pass the custom adapter to the LGV component?

Thanks in advance :)

**Comment by rbuels** [comment, marked as answer]
Yes, JBrowse is designed to support this. You would want to make a plugin that registers a new Adapter, and then you would just want to publish it to the JBrowse plugin store.

### How to do it 

1. start with the [JBrowse plugin template](https://github.com/GMOD/jbrowse-plugin-template), and then add the code for your new adapter.

2. write your adapter like the ones in the JBrowse core code. The JBrowse code is mostly plugins. The code for the core "Sequence" plugin, which contains most of the existing reference sequence adapters, is at https://github.com/GMOD/jbrowse-components/tree/main/plugins/sequence

3. publish it to the plugin store, you would submit a pull request to https://github.com/GMOD/jbrowse-plugin-list

**Reply by garrettjstevens** [reply]
To expand a bit on what @rbuels said, you can see some examples of adding a plugin to the React LGV component here: https://jbrowse.org/storybook/lgv/main/?path=/docs/using-plugins--page. The second example would be what you would use to add a pre-built plugin from the plugin store. Since it sounds like you are using an app bundler, you could instead create the plugin inline like the first example.

The `TwoBitAdapter` in the directory @rbuels linked might be a good place to start, since it is in some ways a bit simpler than some of the other sequence adapters. Also with our core plugins, note that the async `getAdapterClass` when adding the plugin isn't necessary (although it can help with code splitting), so you can use `AdapterClass` instead like this:
```js
pluginManager.addAdapterType(
  () =>
    new AdapterType({
      name: 'TwoBitAdapter',
      displayName: 'TwoBit adapter',
      configSchema,
      AdapterClass: TwoBitAdapter,
    }),
)
```
Also, you can see more docs about adapters here: https://jbrowse.org/jb2/docs/developer_guides/creating_adapter/

**Comment by emiliorighi** [comment]
Hi,

Thank you so much for your response, I just PR the plugin-list repo.

**Comment by derickstark** [comment]
hello everyone can someone help me out with getting a proper API with this file name and content 
Create a file named .env and add the following content there:

USER=Ashutosh
BOTNAME=JARVIS
